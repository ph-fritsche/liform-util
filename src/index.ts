export { isShallowEqual } from './comparison/isShallowEqual'
export { intervalFromString, Interval } from './date/interval'
export { getIndicesOfDescendant } from './dom/indicesOfDescendant'
export { useBoundFunction } from './react/boundFunction'
export { useForkedCallback } from './react/forkedCallback'
export { useForkedRef, updateRef } from './react/forkedRef'
export { useId } from './react/id'
export { ComponentForwardRef, WrapperProps, ForwardRefWrapper, createForwardRefWrapper } from './react/wrapper'
export { isEscaped, countUnescaped, hasUnescaped, indexOfUnescaped } from './string/escaped'
export { filterObject } from './object/filterObject'
