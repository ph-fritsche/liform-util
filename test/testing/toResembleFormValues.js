import '../../src/testing'

const body = document.createElement('body')
body.innerHTML = `
    <form id='testform'>
        <input type='text' name='foo' value='fooValue'/>
        <input type='number' name='bar' value=''/>
        <input type='checkbox' name='baz' value='fooValue' checked/>
        <select multiple name='choice'><option value='foo'/><option selected value='bar'/><option selected value='baz'/></select>
        <select multiple name='choiceB[]'><option value='foo'/><option selected value='bar'/><option selected value='baz'/></select>
    </form>
`

const form = body.querySelector('#testform')

it('Pass for resembling values', () => {
    expect(() => {
        expect(form).toResembleFormValues({
            'foo': true,
            'bar': undefined,
            'baz': true,
            'choice': 'bar,baz',
            'choiceB[0]': true,
            'choiceB[1]': 'baz',
        })
    }).not.toThrowError()

    expect(() => {
        expect(form).toResembleFormValues({
            'foo': 'fooValue',
            'bar': '',
            'baz': 'fooValue',
            'choice': ['bar', 'baz'],
            'choiceB[0]': 'bar',
            'choiceB[1]': 'baz',
        })
    }).not.toThrowError()
})

it('Fail for missing field', () => {
    expect(() => {
        expect(form).toResembleFormValues({
            'foo': 'fooValue',
            'bar': '',
            'baz': 'fooValue',
            'choice': ['bar', 'baz'],
            'choiceB[0]': 'bar',
            'choiceB[1]': 'baz',
            'anotherField': 'fooValue',
        })
    }).toThrowErrorMatchingSnapshot()
})

it('Fail for wrong value', () => {
    expect(() => {
        expect(form).toResembleFormValues({
            'foo': 'fooValue',
            'bar': 0,
            'baz': 'fooValue',
            'choice': ['bar', 'baz'],
            'choiceB[0]': 'bar',
            'choiceB[1]': 'baz',
        })
    }).toThrowErrorMatchingSnapshot()
})

it('Fail for extra field', () => {
    expect(() => {
        expect(form).toResembleFormValues({
            'foo': 'fooValue',
        })
    }).toThrowErrorMatchingSnapshot()
})
