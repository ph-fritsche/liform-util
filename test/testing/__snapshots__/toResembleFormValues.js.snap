// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fail for extra field 1`] = `
"expect(FORM).toResembleFormValues({
  \\"foo\\": \\"fooValue\\"
})

- Expected  -  1
+ Received  + 10

- Object {}
+ Object {
+   \\"bar\\": null,
+   \\"baz\\": \\"fooValue\\",
+   \\"choice\\": Array [
+     \\"bar\\",
+     \\"baz\\",
+   ],
+   \\"choiceB[0]\\": \\"bar\\",
+   \\"choiceB[1]\\": \\"baz\\",
+ }"
`;

exports[`Fail for missing field 1`] = `
"expect(FORM).toResembleFormValues({
  \\"foo\\": \\"fooValue\\",
  \\"bar\\": \\"\\",
  \\"baz\\": \\"fooValue\\",
  \\"choice\\": [
    \\"bar\\",
    \\"baz\\"
  ],
  \\"choiceB[0]\\": \\"bar\\",
  \\"choiceB[1]\\": \\"baz\\",
  \\"anotherField\\": \\"fooValue\\"
})

- Expected  - 3
+ Received  + 1

- Object {
-   \\"anotherField\\": \\"fooValue\\",
- }
+ Object {}"
`;

exports[`Fail for wrong value 1`] = `
"expect(FORM).toResembleFormValues({
  \\"foo\\": \\"fooValue\\",
  \\"bar\\": 0,
  \\"baz\\": \\"fooValue\\",
  \\"choice\\": [
    \\"bar\\",
    \\"baz\\"
  ],
  \\"choiceB[0]\\": \\"bar\\",
  \\"choiceB[1]\\": \\"baz\\"
})

- Expected  - 1
+ Received  + 1

  Object {
-   \\"bar\\": 0,
+   \\"bar\\": null,
  }"
`;
